client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -63,7 +82,32 @@ private void init() {
         } catch (IOException e) {
             throw new RuntimeException("Error reading file " + privateKeyFile, e);
         }
-        var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
-        this.registryApiClient = new RegistryApi(apiClient);
+
+        // temporary to preserve the backwards compatibility
+        if (dataspaceDid.isEmpty()) {
+            var apiClient = createApiClient(service, clientDid, privateKeyData);
+            this.registryApiClient = new RegistryApi(apiClient);
+            return;
+        }
+
+        registryApiClient = new RegistryApi(createApiClient(registrationUrl(), clientDid, privateKeyData));
```

Comment:
I think it would be more understandable to pass the `dataspaceDid.get())` as a param to `registrationUrl()` to make it clear that this case is only reached when there parameter is available. I guess after removing the temporary feature it will be made mandatory.